File: campus-supply-chain/backend/app/schemas/goods.py
from pydantic import BaseModel


class GoodsBase(BaseModel):
    name: str
    category: str = ""
    spec: str = ""
    unit: str = ""
    safety_level: str = "medium"
    shelf_life_days: int = 365


class GoodsCreate(GoodsBase):
    pass


class GoodsUpdate(BaseModel):
    name: str | None = None
    category: str | None = None
    spec: str | None = None
    unit: str | None = None
    safety_level: str | None = None
    shelf_life_days: int | None = None


class GoodsResponse(GoodsBase):
    id: int

    class Config:
        from_attributes = True

File: campus-supply-chain/frontend/src/api/goods.ts
import request from './request'

export interface GoodsItem {
  id: number
  name: string
  category: string
  spec: string
  unit: string
  safety_level: string
  shelf_life_days: number
}

export function listGoods(params?: { keyword?: string; category?: string }) {
  return request.get<GoodsItem[]>('/goods', { params })
}

export function createGoods(data: Partial<GoodsItem>) {
  return request.post<GoodsItem>('/goods', data)
}

export function updateGoods(id: number, data: Partial<GoodsItem>) {
  return request.put<GoodsItem>(`/goods/${id}`, data)
}

export function deleteGoods(id: number) {
  return request.delete(`/goods/${id}`)
}
